packages/core/src/utils/migration.ts

```diff
@@ -0,0 +1,190 @@
+/**
+ * @license
+ * Copyright 2025 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as fs from 'fs';
+import * as path from 'path';
+import * as os from 'os';
+import { GEMINI_DIR } from './paths.js';
+
+const INTERNAL_DIR = 'internal';
```

Comment:
We actually have a `tmp` directory that's also exposed on the config. How would you feel about moving all the contents there?